Off-by-one in SME

When handling unexpected discoveries, there appears to be an off-by-one error in the SME that eventually tries to advance a chain that has no more mutations.  Results in:

```
08:58:09.303:kraken:DEBUG:StateMutationEngine:123e4567-e89d-12d3-a456-425455440096 found a new path
panic: runtime error: index out of range

goroutine 24 [running]:
github.com/hpc/kraken/core.(*StateMutationEngine).advanceMutation(0xc0001c6000, 0xc0000c73e0, 0x24, 0xc000394b80)
	/home/vagrant/go/src/github.com/hpc/kraken/core/StateMutationEngine.go:730 +0x61a
github.com/hpc/kraken/core.(*StateMutationEngine).handleUnexpected(0xc0001c6000, 0xc0000c73e0, 0x24, 0xc0000c7405, 0x9, 0x9ee0e0, 0xc00041e6e8, 0x185)
	/home/vagrant/go/src/github.com/hpc/kraken/core/StateMutationEngine.go:818 +0xded
github.com/hpc/kraken/core.(*StateMutationEngine).updateMutation(0xc0001c6000, 0xc0000c73e0, 0x24, 0xc0000c7405, 0x9, 0xab7c60, 0xc00038f260, 0x99)
	/home/vagrant/go/src/github.com/hpc/kraken/core/StateMutationEngine.go:878 +0x1175
github.com/hpc/kraken/core.(*StateMutationEngine).handleEvent(0xc0001c6000, 0xb771a0, 0xc0001dca50)
	/home/vagrant/go/src/github.com/hpc/kraken/core/StateMutationEngine.go:971 +0x336
github.com/hpc/kraken/core.(*StateMutationEngine).Run(0xc0001c6000)
	/home/vagrant/go/src/github.com/hpc/kraken/core/StateMutationEngine.go:286 +0x6c8
created by github.com/hpc/kraken/core.(*Kraken).Run
	/home/vagrant/go/src/github.com/hpc/kraken/core/Kraken.go:198 +0x10a
got stream read error on mutation stream: rpc error: code = Unavailable desc = transport is closing
got stream read error on mutation stream: rpc error: code = Unavailable desc = transport is closing
got stream send error on logger stream: EOF
got stream send error on logger stream: EOF
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,6 @@
 .vscode
-./build
+/build
 tmp
 munge.key
 ssh_host_*key*
-./kraken-build
+/kraken-build

--- a/core/StateMutationEngine.go
+++ b/core/StateMutationEngine.go
@@ -795,7 +795,7 @@ func (sme *StateMutationEngine) handleUnexpected(node, url string, val reflect.V
 		sme.Logf(DEBUG, "%s is devolving back %d steps due to an unexpected regression", node, m.cur-i)
 
 		n.SetValues(rewind)
-		m.chain = append(m.chain[:m.cur], m.chain[i-1:]...)
+		m.chain = append(m.chain[:m.cur+1], m.chain[i:]...)
 		sme.advanceMutation(node, m)
 		return
 	}
@@ -814,7 +814,7 @@ func (sme *StateMutationEngine) handleUnexpected(node, url string, val reflect.V
 		}
 		// update the chain & increment
 		sme.Logf(DEBUG, "%s found a new path", node)
-		m.chain = append(m.chain[:m.cur], p.chain...)
+		m.chain = append(m.chain[:m.cur+1], p.chain...)
 		sme.advanceMutation(node, m)
 		return
 	}

--- a/utils/layer0/uinit/uinit.go
+++ b/utils/layer0/uinit/uinit.go
@@ -49,17 +49,17 @@ const (
 func loadModules() (cmds []command, e error) {
 	var f *os.File
 	if _, e = os.Stat(moduleFile); os.IsNotExist(e) {
-		fmt.Printf("%s does not exist, we will not load modules", moduleFile)
+		fmt.Printf("%s does not exist, we will not load modules\n", moduleFile)
 		return
 	}
 	if f, e = os.Open(moduleFile); e != nil {
-		fmt.Printf("failed to open module file, %s: %v", moduleFile, e)
+		fmt.Printf("failed to open module file, %s: %v\n", moduleFile, e)
 		return
 	}
 	defer f.Close()
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
-		fmt.Printf("adding module to load list: %s", scanner.Text())
+		fmt.Printf("adding module to load list: %s\n", scanner.Text())
 		cmd := command{
 			Cmd:        insmodCmd,
 			Args:       []string{insmodCmd, scanner.Text()},
@@ -72,7 +72,7 @@ func loadModules() (cmds []command, e error) {
 
 func goExec(cmd *exec.Cmd) {
 	if err := cmd.Run(); err != nil {
-		log.Printf("command %s failed: %s", cmd.Path, err)
+		log.Printf("command %s failed: %s\n", cmd.Path, err)
 	}
 }
 

